### PR TITLE
120 gsd error metrics throw if reference did not detect a gs

### PR DESCRIPTION
**Pull request by a-mosquito**

### CODE REVIEW

**Comment by a-mosquito** [comment]
I added a `zero_division_hint` parameter to the `calculate_unmatched_gsd_performance_metrics` function. I named it differently than the parameter used for the basic evaluation functions as it behaves differently
- set to `False`: warning is suppressed and values are set to nan
- set to "warn": warning is raised and values are set to nan
- set to "raise": `ZeroDivisionError` is raised

(comp. to `zero_division` parameter from f1-score etc: result is either set to 0, 1, or warning is raised)

File: mobgap/gsd/evaluation.py [hunk 1]
```diff
@@ -142,6 +147,11 @@ def calculate_unmatched_gsd_performance_metrics(
        Should have the same format as `gsd_list_detected`.
     sampling_rate_hz
         Sampling frequency of the recording in Hz.
+    zero_division_hint : "warn", "raise" or False, default="warn"
+        Controls the behavior when there is a zero division. If set to "warn",
+        affected metrics are set to NaN and a warning is raised.
+        If set to "raise", a ZeroDivisionError is raised.
+        If set to False, the warning is suppressed and the affected metrics are set to NaN.
```
Comment (AKuederle) [inline mobgap/gsd/evaluation.py:154]: Can we have the option to use "NaN" instead of False (so to make it consistent with the other option)

Comment (AKuederle) [reply mobgap/gsd/evaluation.py:154]: And can you specify in what scope that will be applied. Very specifically we safeguard against the reference_gs_duration_s == 0

**Review by AKuederle — CHANGES_REQUESTED** [review]
Small wording comment. Other than that ready to merge

File: mobgap/gsd/evaluation.py (hunk 1 above)
Comment (AKuederle) [reply mobgap/gsd/evaluation.py]: Sorry I wasn't clear on that one. Can we actually use the value np.nan? In the other case, when we put in a 1 th output will be 1. So here it seems natural if we put np.nan, we get np.nan. Or is that a bad idea?

**Review by AKuederle — APPROVED** [review]

**Pull request merged**